residue_genomic_position_script.py

```diff
@@ -0,0 +1,266 @@
+import argparse
+from email import header
+from json import JSONDecodeError
+import logging
+import psutil
+
+import pandas as pd
+import requests
+from pandarallel import pandarallel
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, IntegerType, StringType
+
+
+# Global configuration for Spark and Pandarallel.
+spark = SparkSession.builder.master('local[*]').getOrCreate()
+
+pandarallel.initialize(
+    nb_workers=psutil.cpu_count(),
+    progress_bar=True,
+)
+
+def main():
+
+    # PLIP INPUT (contain wanted data)
+    plip_json_input = (
+        # "gene_mapped_structures.json"
+        spark.read.json(args.plip_input)
+
+        .select("pdbStructureId", "chains", f.explode("compoundIds").alias("pdbCompoundId"))
+
+        .select("pdbStructureId", "pdbCompoundId", f.explode("chains").alias("chains"))
+
+        )
+
+    plip_json_input_v2 = (plip_json_input
+
+                        .withColumn("chainId", plip_json_input["chains.chainId"])
+
+                        .withColumn("geneId", plip_json_input["chains.geneId"])
+
+                        .withColumn("uniprotId", plip_json_input["chains.uniprot"])
+
+                        .drop("chains")
+
+                        )
```

Comment:
There are a few comments here:
* If a variable is not used anywhere (`plip_json_input`), but immediately processed further, you can just join the processes together.
* Try to add informative variable names. And please avoid such names as `_v2`, because it is very confusing.
* Try to apply consistent indentation. (somehow `plip_json_input_v2` has more whitespace.)
* Try to not add excessive whitespaces/new lines. It makes code less readable.
* You can extract all fields from a struct using the `*` wildcard.

So I would combine the two steps as following (note that the compound id is not required):

```suggestion
    plip_json_input = (
        spark.read.json(args.plip_input)
        # Explode chains and extract chain related fields into columns:
        .withColumn("chain", f.explode("chains"))
        .select(f.col("pdbStructureId"), f.col("chain.*"))
        .persist()
    )
```